### PR TITLE
Add a helper script for identifying the SDK release containing a SHA

### PR DESCRIPTION
**Pull request by baronfel**
While researching commit history for https://github.com/dotnet/docs/pull/29354 I created a script that will scrape the tag list and walk it in chronological order, stopping when it finds the first tag that contains the provided commit-ish reference.  

I think it's a difficult enough task that keeping this script would be useful.

### CODE REVIEW

**Review by joeloff — APPROVED** [review]
Sounds great. We'll probably need a .sh copy too.

**Comment by KalleOlaviNiemitalo** [comment]
`git describe --contains` is similar.

File: scripts/find-earliest-tag.ps1 [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        }
+    }
+    return $tags.Keys.GetEnumerator()
+}
+
+$allTags=(Get-Tags "upstream")
+$found=$false
+foreach ($tag in $allTags) {
+    if (Check-Ref $targetSHA $tag) {
+        Write-Host "The earliest tag that is an ancestor of $targetSHA is $tag" -ForegroundColor Green
```
Comment (KalleOlaviNiemitalo) [inline scripts/find-earliest-tag.ps1:46]: Don't you mean "descendant"? As in, the commit was made first, and the tag later.

Comment (baronfel) [reply scripts/find-earliest-tag.ps1:46]: yeah, that's incorrect on my part

**Comment by baronfel** [comment]
The pointer to `git describe` was a good one, @KalleOlaviNiemitalo - the invocation we want would be something like `git describe --tags --contains <target SHA> --exclude "v*-*"` to get similar behavior, _and_ it's way faster that walking each SHA.

**Comment by KalleOlaviNiemitalo** [comment]
There's also `git tag --contains` if you don't want decorations, but then it would list all matching tags rather than just the earliest one.

Do you run your script manually or as part of some automation?

**Comment by baronfel** [comment]
This is purely for manual fact finding. The motivation in this case was finding out which SDK releases started tracking certain telemetry data points that slipped through our documentation (which have now been documented :) ).

**Comment by baronfel** [comment]
Closing because it's a one-liner!

**Pull request closed without merging**